pkg/metrics/instance/batchapplyerror.go

```diff
@@ -0,0 +1,63 @@
+package instance
+
+import "fmt"
+
+// BatchApplyError contains the failed configs with error, successful configurations, and then if there is an error
+// that is not tied to a specific config then NonConfigError contains it
+type BatchApplyError struct {
+	Failed []BatchFailure
+	// NonConfigError is used when the error is not with a specific configuration, but likely the grouping
+	// or applying the group.
+	NonConfigError error
+}
+
+// BatchFailure for a given config has the associated error
+type BatchFailure struct {
+	Err    error
+	Config Config
+}
+
+// CreateBatchApplyErrorOrNil will create an error if failed has > 0 elements on nonConfigError is not nil, else will
+// return nil
+func CreateBatchApplyErrorOrNil(failed []BatchFailure, nonConfigError error) error {
```

Comment:
Given my comment w/r/t the struct, we should be able to remove this function an inline the call. At the very least, this shouldn't be an exposed function anyway.